### PR TITLE
fix(groups): Add isMember method to chat content module

### PR DESCRIPTION
**Pull request by borismelnik**
Closes: #4615

### What does the PR do

Add `isMember` method to chat content module 

### Affected areas

Groups / ChatContentModule

### CODE REVIEW

**Comment by alaibe** [comment]
Please see: https://github.com/status-im/status-desktop/pull/4661 as it creates a dedicated chat member module and why

**Comment by borismelnik** [comment]
> Please see: #4661 as it creates a dedicated chat member module and why

@alaibe i understand your point but its just PR. We can close my when your changes will be added.

**Comment by alaibe** [comment]
> @alaibe i understand your point but its just PR

Not sure to understand that here, I am only giving pointer to a discussion in progress regarding the group subject I think you might be interesting at looking

File: ui/app/AppLayouts/Chat/popups/GroupInfoPopup.qml [hunk 1]
```diff
@@ -130,7 +130,7 @@ StatusModal {
         }
 
         NoFriendsRectangle {
-            visible: popup.store.addToGroupContacts.count === 0 && memberCount > 0
+            visible: chatSectionModule.listOfMyContacts.rowCount() === 0 && memberCount > 0
```
Comment (Khushboo-dev-cpp) [inline ui/app/AppLayouts/Chat/popups/GroupInfoPopup.qml:133]: can we use the qml stores please?

File: src/app/modules/main/chat_section/module.nim [hunk 2]
```diff
@@ -282,31 +282,11 @@ method chatContentDidLoad*(self: Module) =
...
-      return
-
-  # here, in this step we have appropriate item and subitem assigned
-  self.setActiveItemSubItem(item.BaseItem.id, subItemId)
-
 method activeItemSubItemSet*(self: Module, itemId: string, subItemId: string) =
   let item = self.view.chatsModel().getItemById(itemId)
   if(item.isNil):
     # Should never be here
-    error "chat-view unexisting item id: ", itemId, methodName="activeItemSubItemSet"
```
Comment (Khushboo-dev-cpp) [inline src/app/modules/main/chat_section/module.nim:309]: why did we remove method names? just curious >=

**Review by Khushboo-dev-cpp — APPROVED** [review]
LGTM :)

**Review by jrainville — COMMENTED** [review]
Like Anthony said, he's already working on a new members module or an improvement on the users module. Once that's done, the function to check if someone is  a member should be done from there.

As you can see, there is already a function in the model to check if a pubkey is in the list: https://github.com/status-im/status-desktop/blob/master/src/app/modules/shared_models/user_model.nim#L119

Currently, that usersModule is flawed, because in the case of group chats, it does not reflect the current members accurately, but it is planned to be fixed.

Once it is fixed, you can make that function a slot, or add a new slot that calls that function (with a better name) and use it instead.

File: src/app/modules/main/chat_section/chat_content/module.nim [hunk 3]
```diff
@@ -205,6 +205,13 @@ method getMyChatId*(self: Module): string =
 method isMyContact*(self: Module, contactId: string): bool =
   self.controller.getMyAddedContacts().filter(x => x.id == contactId).len > 0
 
+method isMember*(self: Module, contactId: string): bool =
```
Comment (saledjenic) [inline src/app/modules/main/chat_section/chat_content/module.nim:208]: @borismelnik checking if contact with a certain id is member of this chat or not, from my point of view is totally fine  to check that in a way you did here, since `chatService` (`chats` table maintained there) is our source of truth in terms of everything chat related. All our models including users list are fulfilled from data coming from different services.

File: src/app/modules/main/chat_section/module.nim [hunk 4]
```diff
@@ -282,31 +282,11 @@ method chatContentDidLoad*(self: Module) =
 method setActiveItemSubItem*(self: Module, itemId: string, subItemId: string) =
   self.controller.setActiveItemSubItem(itemId, subItemId)
 
-method makeChatWithIdActive*(self: Module, chatId: string) =
```
Comment (saledjenic) [inline src/app/modules/main/chat_section/module.nim:285]: @borismelnik I am wondering why did you remove this method? This is very important method for us. Please revert it.

File: src/app/modules/main/chat_section/module.nim (hunk 2 above)
Comment (saledjenic) [inline src/app/modules/main/chat_section/module.nim:309]: @borismelnik this line is also very meaningful, please revert it.

**Review by saledjenic — CHANGES_REQUESTED** [review]
@borismelnik what do you have against `methodName` in errors/debug info? :) That provides us more info so we may easier follow what's happening. If you don't have a very strong reason to remove it, please revert them.

File: src/app/modules/shared_models/user_model.nim [hunk 5]
```diff
@@ -124,7 +124,7 @@ QtObject:
     self.endRemoveRows()
     self.countChanged()
 
-  proc isContactWithIdAdded*(self: Model, id: string): bool = 
+  proc isContactWithIdAdded*(self: Model, id: string): bool {.slot.} =
```
Comment (saledjenic) [inline src/app/modules/shared_models/user_model.nim:127]: @borismelnik exposing proc directly from the model, as it's done here, is against the rules we're trying to establish. Please revert this and add appropriate method to the `view` of the `module` where you need this info, and get what you need that way.

File: ui/app/AppLayouts/Chat/panels/ContactListPanel.qml [hunk 6]
```diff
@@ -31,14 +32,15 @@ ScrollView {
             image.isIdenticon: !!model.identicon
             visible: {
                 if (selectMode) {
-                    return !searchString || model.name.toLowerCase().includes(searchString)
+                    return (!searchString || model.name.toLowerCase().includes(searchString))
+                            && !chatContentModule.usersModule.model.isContactWithIdAdded(model.pubKey)
```
Comment (saledjenic) [inline ui/app/AppLayouts/Chat/panels/ContactListPanel.qml:36]: @borismelnik this check here `!chatContentModule.usersModule.model.isContactWithIdAdded(model.pubKey)` causes other type of issue cause it will alway use list of users added to the active chat. Please fix this.

**Review by saledjenic — CHANGES_REQUESTED** [review]
@borismelnik it seems we need to handle multiple things here. You see that `ContactListPanel` is used form the 2 places, and there are multiple cases you should cover:
- create group chat if there are no chats added at all
- create group chat regardless what chat is the selected one
- view group chat details from clicking on the chat toolbar
- view group chat details regardless what chat is the selected one (right click "View Group" option)

the issue I currently see is that `GroupInfoPopup` and `GroupChatPopup` relies on the currently selected chat, what is not what we want. They should be independent of selected chat.

**Comment by borismelnik** [comment]
> @borismelnik it seems we need to handle multiple things here. You see that `ContactListPanel` is used form the 2 places, and there are multiple cases you should cover:
> 
> * create group chat if there are no chats added at all
> * create group chat regardless what chat is the selected one
> * view group chat details from clicking on the chat toolbar
> * view group chat details regardless what chat is the selected one (right click "View Group" option)
> 
> the issue I currently see is that `GroupInfoPopup` and `GroupChatPopup` relies on the currently selected chat, what is not what we want. They should be independent of selected chat.

i understand your point, but i will fix it in other PR. Cos i think the big refactoring part if out of this bug.

**Comment by anastasiyaig** [comment]
@borismelnik why the of group chat members is scrollable when having 1 member and the only participant is in the middle of the screen?


https://user-images.githubusercontent.com/82375995/153386391-668b9d60-83e9-4bab-8054-50892cb71bc3.mov
